WooCommerce/Classes/ViewRelated/AppCoordinator.swift

```diff
@@ -66,6 +69,14 @@ private extension AppCoordinator {
     /// Displays logged in tab bar UI.
     ///
     func displayLoggedInUI() {
+        // if the previous role check indicates that the user is ineligible, then we want to show the error message.
+        // Eligibility status is stored locally in user defaults, and updated asynchronously upon app launch.
+        if let errorInfo = roleEligibilityUseCase.lastEligibilityErrorInfo(),
```

Comment:
⚠️  I think this is a bit missleading. A use case is supposed to serve a single purpose, but this use-case already has two. `checkEligibility` and `lastEligibilityErrorInfo`. What do you think about moving `lastEligibilityErrorInfo()` to `AppSettingsStore`?

Reply:
> A use case is supposed to serve a single purpose,

Ah, I didn't know that we are supposed to have only one public method per use case. I thought it's more of being responsible for a single context or object, but still allowing more than one method. I'll try to separate the responsibility to another object, then 👍 

> moving lastEligibilityErrorInfo() to AppSettingsStore?

Good idea! Will do this soon. 👍

Reply:
Update: comment addressed in #4492, particularly in [this commit](https://github.com/woocommerce/woocommerce-ios/pull/4492/commits/f2e1cccf92c216a97dcf4ab30806c9024fd2330e)! 🙂

Reply:
This is addressed in #4665!